ledger/acctupdates.go

```diff
@@ -1473,6 +1483,72 @@ func (au *accountUpdates) upgradeDatabaseSchema3(ctx context.Context, tx *sql.Tx
 	return 4, nil
 }
 
+
+// upgradeDatabaseSchema4 upgrades the database schema from version 4 to version 5,
+// cleaning old empty catchpoint directories.
+func (au *accountUpdates) upgradeDatabaseSchema4(ctx context.Context, tx *sql.Tx) (updatedDBVersion int32, err error) {
+	//validate no empty dirs
+	err = removeEmptyDirsOnSchemaUpgrade()
+	if err != nil {
+		return 0, err
+	}
+
+	// update version
+	_, err = db.SetUserVersion(ctx, tx, 5)
+	if err != nil {
+		return 0, fmt.Errorf("accountsInitialize unable to update database schema version from 4 to 5: %v", err)
+	}
+	return 5, nil
+}
+
+func  removeEmptyDirsOnSchemaUpgrade() (err error) {
+	f, err := os.Open(catchpointDirName)
```

Comment:
The starting point isn't `catchpointDirName`, since that directory is relative to the CWD. instead, you should use the `au.dbDirectory` and use it in the same way it's being used above.

Reply:
Done